llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp

```diff
@@ -484,6 +484,7 @@ namespace {
     SDValue visitCTTZ_ZERO_UNDEF(SDNode *N);
     SDValue visitCTPOP(SDNode *N);
     SDValue visitSELECT(SDNode *N);
+    SDValue visitCTSELECT(SDNode *N);
```

Comment:
I hate to say it, but is it possible that the naming of this node will cause confusion? Two lines above this `visitCTSELECT` I see the existing `visitCTPOP`, which _looks_ very similar, but is totally different – your CT stands for "constant time" but theirs stands for "count".